### PR TITLE
Fixed #6734: ActiveForm validation js 'input' and '.field-' container cl...

### LINKED ISSUE #6734: ActiveField: Setting the `id` for an input field does not change the `id` for the validation js
I had a situation where I was using an `ActiveForm` instance for a batch input form and needed to change the fields to use custom `ids`. The initial javascript generated per input was as follows:

```javascript
jQuery('#w0').yiiActiveForm([{
    "id": "vehicle-type_id",
    "name": "[]type_id",
    "container": ".field-vehicle-type_id",
    "input": "#vehicle-type_id",
    "validate": function(attribute, value, messages, deferred) {
        yii.validation.required(value, messages, {
            "message": "Type cannot be blank."
        });
        yii.validation.number(value, messages, {
            "pattern": /^\s*[+-]?\d+\s*$/,
            "message": "Type must be an integer.",
            "skipOnEmpty": 1
        });
    }
}, {
```
This resulted in all the `type_id` fields being validated together since the `id`s were the same.

On changing the code to include the id as follows:

```php
$form->field($model, '[]type_id')->dropdown($items, ['id' => 'mycustomid']);
```

the validation javascript was unchanged.

### PR DESCRIPTION
**Pull request by cebe**
...ass now uses selectors['input']

fixes #6734

### CODE REVIEW

File: framework/widgets/ActiveField.php [hunk 1]
```diff
@@ -800,4 +800,12 @@ protected function getClientOptions()
             'error' => '.help-block',
         ]);
     }
+    
+    /**
+     * Returns the input id from [[selectors]] or [[Html::getInputId()]]
+     * @return string the input id
+     */
+	protected function getInputId() {
```
Comment (qiangxue) [inline framework/widgets/ActiveField.php]: code style?

**Comment by qiangxue** [comment]
hmm... the change looks strange to me: the `id` for each field in js code is used to get the corresponding error message. It is different from selector ID.

**Comment by kanyuga** [comment]
@qiangxue the other option would be to allow `selectors['id']` to be edited manually. Thoughts?

**Comment by SilverFire** [comment]
Resolved by https://github.com/yiisoft/yii2/commit/13fca3bc54e22e9e89f9b86eeedbfb87a248d750

**Pull request closed without merging**